sdk/storage/Azure.Storage.Common/tests/Shared/StorageTestBase.cs

```diff
@@ -470,5 +471,20 @@ protected async Task<Stream> CreateLimitedMemoryStream(long size, long maxMemory
             }
             return stream;
         }
+
+#pragma warning disable SA1402 // File may only contain a single type
+    private class TestCredential : TokenCredential
```

Comment:
Nit - can we rename this to `StorageTestTokenCredential`?  We've got about 10 of these in the repo and it'd make it a little easier to search on github.